File: 命令模式(旅行社项目开发)/Invoker.java
//原有的基础上增加Invoker类,作用是根据客户的命令安排不同的组员进行工作
public class Invoker{
    //什么命令
    private Command command;
    //客户发出命令
    public void setCommand(Command _command){
        this.command = _command;
    }

    //执行客户命令
    public void action(){
        this.command.execute();
    }
}

File: 通用代码/命令模式(Reciver打开)(C++)/Invoker.cpp
#include "Invoker.h"

Invoker::Invoker(void)
{
    cout << "Invoker..." << endl;
}

Invoker::~Invoker(void)
{
    cout << "~Invoker..." << endl;
}

void Invoker::SetCommand(Command* pCommand)
{
    this->_cmd = pCommand;
}

void Invoker::Action()
{
    this->_cmd->Execute();
}
